app/models/nic/interface.rb

```diff
@@ -105,7 +105,7 @@ def normalize_name
       # no hostname was given or a domain was selected, since this is before validation we need to ignore
       # it and let the validations to produce an error
       return if name.empty?
-      if domain.nil? && name.include?('.') && changed_attributes['domain_id'].blank?
+      if domain_id.nil? && name.include?('.') && changed_attributes['domain_id'].blank?
         # try to assign the domain automatically based on our existing domains from the host FQDN
         self.domain = Domain.find_by(:name => name.partition('.')[2])
```

Comment:
Souldn't this and the below `Domain.find` also use `.unscoped` to ensure the search isn't scoped?

Reply:
We could do that, but is that better?  Isn't the presence of domain_id enough?  I guess the .unscoped is more explicit which is better.  Will update!